src/contract/polls.cpp

```diff
@@ -0,0 +1,627 @@
+
+    if (sTitle.empty() || sQuestion.empty() || sAnswers.empty() || sURL.empty())
+    {
+        return std::make_pair("Error", "Must specify a poll title, question, answers, and URL\n");
+    }
+    else if (days < 7)
+        return std::make_pair("Error", "Minimum duration is 7 days; please specify a longer poll duration.");
+    else
+    {
+        double nBalance = GetTotalBalance();
+        if (nBalance < 100000)
+            return std::make_pair("Error", "You must have a balance > 100,000 GRC to create a poll.  Please post the desired poll on https://cryptocurrencytalk.com/forum/464-gridcoin-grc/ or https://github.com/Erkan-Yilmaz/Gridcoin-tasks/issues/45");
+        else
+        {
+            if (iSharetype != 1 && iSharetype != 2 && iSharetype != 3 && iSharetype != 4 && iSharetype != 5)
+                return std::make_pair("Error", "You must specify a value of 1, 2, 3, 4 or 5 for the sSharetype.");
+            else
+            {
+                std::string expiration = RoundToString(GetAdjustedTime() + (days*86400), 0);
+                std::string contract = "<TITLE>" + sTitle + "</TITLE><DAYS>" + std::to_string(days) + "</DAYS><QUESTION>" + sQuestion + "</QUESTION><ANSWERS>" + sAnswers + "</ANSWERS><SHARETYPE>" + std::to_string(iSharetype) + "</SHARETYPE><URL>" + sURL + "</URL><EXPIRATION>" + expiration + "</EXPIRATION>";
+                std::string result = AddContract("poll", sTitle, contract);
+                return std::make_pair("Success",result);
+            }
+        }
+    }
+}
+
+std::pair<std::string, std::string> CreateVoteContract(std::string sTitle, std::string sAnswer)
+{
+    if (sTitle.empty() || sAnswer.empty())
+    {
+        return std::make_pair("Error", "Must specify a poll title and answers\n");
+    }
+
+    if (pwalletMain->IsLocked())
+        return std::make_pair("Error", "Please fully unlock the wallet first.");
+    else if (fWalletUnlockStakingOnly)
+        return std::make_pair("Error", "Wallet is unlocked for staking only, needs to be fully unlocked for voting");
+    else if (WalletOutOfSync())
+        return std::make_pair("Error", "Wallet is not synced, you need to be synced in order to place a vote");
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    //Verify the Existence of the poll, the acceptability of the answer, and the expiration of the poll: (EXIST, EXPIRED, ACCEPTABLE)
+    //If polltype == 1, use magnitude, if 2 use Balance, if 3 use hybrid:
+    //6-20-2015
+    double nBalance = GetTotalBalance();
+    uint256 hashRand = GetRandHash();
+    std::string email = GetArgument("email", "NA");
+    boost::to_lower(email);
+    GlobalCPUMiningCPID.email = email;
+    GlobalCPUMiningCPID.lastblockhash = GlobalCPUMiningCPID.cpidhash;
+    if (!PollExists(sTitle))
+        return std::make_pair("Error", "Poll does not exist.");
+    else
+    {
+        if (PollExpired(sTitle))
+            return std::make_pair("Error", "Sorry, Poll is already expired.");
+        else
```

Comment:
I know you just moved the functions, but can you drop the else on this one to reduce one depth?